v0/destinations/hs/transform.js

```diff
@@ -292,22 +299,31 @@ function batchEvents(destEvents) {
     let batchEventResponse = defaultBatchRequestConfig();
 
     chunk.forEach(ev => {
-      const { email, updatedProperties } = getEmailAndUpdatedProps(
-        ev.message.body.JSON.properties
-      );
-      ev.message.body.JSON.properties = updatedProperties;
-      identifyResponseList.push({
-        email,
-        properties: ev.message.body.JSON.properties
-      });
-      metadata.push(ev.metadata);
+      const { properties } = ev.message.body.JSON
+      if (properties && !(properties instanceof Array)) {
```

Comment:
```suggestion
      if (properties && !Array.isArray(properties)) {
```
Generally, `Array.isArray()` performs more robust checking

Reply:
done